docs/en/query_language/functions/other_functions.md

```diff
@@ -607,38 +607,55 @@ SELECT defaultValueOfArgumentType( CAST(1 AS Nullable(Int8) ) )
 └───────────────────────────────────────────────────────┘
 ```
 
-## indexHint
+## indexHint {#indexhint}
 
-Outputs data in the range selected by the index without filtering by the expression specified as an argument.
+Outputs all data from the range that contains data where expression is true.
```

Comment:
* Actually the previous description was more clear
* Also needs a list of table engines that are supported and what happens if it isn't
* Needs some examples of usecases why/when this might be useful

Reply:
Description returned, the rest is very long to write